### PR TITLE
Added support for submitting problems through submit endpoint

### PR DESCRIPTION
**Pull request by AlannaFey**
It says there are a lot of changes because I had to move everything from the original code into the uid block to make sure that the user has a valid user id.  Other than that now if you want to submit something for a problem, you include source_code, problem_id, uid, and language_id.  Arbitrary submissions should remain the same.
image
image

### CODE REVIEW

File: functions/src/judge.ts [hunk 1]
```diff
@@ -33,109 +34,191 @@ export async function judge_is_online(_req: Request, res: Response) {
...
+    if (outputs == undefined) {
+      missing.push('Missing outputs array')
+    }
+    if (missing.length > 0) {
+      return res.status(400).json({ error: missing })
+    }
+  } else {
+    // problem submission
+    inputs = []
+    outputs = []
```
Comment (AsirAAlam) [inline functions/src/judge.ts:106]: is this necessary?

**Comment by AsirAAlam** [comment]
Should we update the comments on `index.ts` for documenting this?

File: functions/src/judge.ts [hunk 2]
```diff
@@ -33,109 +34,191 @@ export async function judge_is_online(_req: Request, res: Response) {
   }
 }
 
+/* eslint-disable  @typescript-eslint/no-explicit-any */
```
Comment (cameroncuster) [inline functions/src/judge.ts:37]: it's preferred to use unknown and cast

**Review by cameroncuster — APPROVED** [review]
Went through this pretty quickly, but I think it looks ok

File: functions/src/judge.ts (hunk 1 above)
Comment (AlannaFey) [reply functions/src/judge.ts:106]: yes, in case we are doing a problem submission, the inputs/outputs array should be undefined.

**Pull request merged**